homeassistant/components/meteoalarm/binary_sensor.py

```diff
@@ -37,43 +38,88 @@
         vol.Required(CONF_COUNTRY): cv.string,
         vol.Required(CONF_PROVINCE): cv.string,
         vol.Optional(CONF_LANGUAGE, default="en"): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_NAME, default=DOMAIN): cv.string,
     }
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the MeteoAlarm binary sensor platform."""
 
-    country = config[CONF_COUNTRY]
-    province = config[CONF_PROVINCE]
-    language = config[CONF_LANGUAGE]
-    name = config[CONF_NAME]
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=config,
+    )
+    if (
+        result["type"] is FlowResultType.CREATE_ENTRY
+        or result["reason"] == "single_instance_allowed"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.12.0",
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "MeteoAlarm",
+            },
+        )
+        return
+    async_create_issue(
+        hass,
+        DOMAIN,
+        f"deprecated_yaml_import_issue_{result['reason']}",
+        breaks_in_ha_version="2024.12.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key=f"deprecated_yaml_import_issue_{result['reason']}",
+        translation_placeholders={"domain": DOMAIN, "integration_title": "MeteoAlarm"},
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up MeteoAlarm from config_entry."""
 
     try:
-        api = Meteoalert(country, province, language)
+        api = Meteoalert(
+            entry.data[CONF_COUNTRY],
+            entry.data[CONF_PROVINCE],
+            entry.data[CONF_LANGUAGE],
+        )
     except KeyError:
-        _LOGGER.error("Wrong country digits or province name")
+        LOGGER.error("Wrong country digits or province name")
         return
```

Comment:
This should be moved to `__init__.py`. We should store that in `entry.runtime_data` (please make sure you extend the typing of ConfigEntry for type checking

Reply:
@joostlek Thanks for looking at this so quick. 

This is my first config-flow conversion.
I've been having some trouble finding documentation for types such as ConfigEntry, ConfigFlowResult, etc. Is there anything available that you know of, besides using the source?

Reply:
The source and all the other integrations out there. Reworking the docs is something on my list

Reply:
AirGradient has an example, `type MeteoAlarmConfigEntry = ConfigEntry[...]`